lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -86,10 +86,11 @@ def run_qemu
 
       def run_vm
         @machine.dump_config
-        @machine.run_pre_start_commands
 
         @qemu_thread = Thread.new do
           loop do
+            @machine.run_pre_start_commands
```

Comment:
`@machine.run_pre_start_commands` should be in `Thread.handle_interrupt(Object => :on_blocking)` block. There is a possibility that `@machine.run_pre_start_commands` did some work that must be discarded with `              @machine.run_post_stop_commands` before raising an exception.